### PR TITLE
Xgboost predict

### PR DESCRIPTION
**Pull request by typhoonzero**
part of work: https://github.com/sql-machine-learning/sqlflow/issues/749

### CODE REVIEW

File: sql/template_xgboost.go [hunk 1]
```diff
@@ -58,22 +58,20 @@ feature_specs["{{$value.FeatureName}}"] = {
...
-				for item in gen():
-						features, label = item
-						row_data = [str(label[0])] + ["%d:%f" % (i, v) for i, v in enumerate(features)]
-						f.write("\t".join(row_data) + "\n")
-		# TODO(yancey1989): genearte group and weight text file if necessary
-		return xgb.DMatrix(fn) 
-
-dtrain = xgb_dataset('train.txt', "{{.TrainingDatasetSQL}}")
-dtest = xgb_dataset('test.txt', "{{.ValidationDatasetSQL}}")
+    gen = db_generator(driver, conn, session_cfg, dataset_sql, feature_column_names, "{{.Y.FeatureName}}", feature_specs)
```
Comment (Yancey0623) [inline sql/template_xgboost.go:64]: Thanks for fixing the indent.

File: sql/codegen_xgboost_test.go [hunk 2]
```diff
@@ -51,3 +58,17 @@ func TestXGBFiller(t *testing.T) {
 	a.NoError(err)
 	a.Equal(filler.ParamsCfgJSON, string(paramsJSON))
 }
+
+func TestXGBFillerPredict(t *testing.T) {
```
Comment (Yancey0623) [inline sql/codegen_xgboost_test.go:67]: Please also add a unit test in `executor_test` to test the pred SQL.

Comment (typhoonzero) [reply sql/codegen_xgboost_test.go:67]: Done.

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Pull request merged**